action/src/stoatApiHelpers.ts

```diff
@@ -5,10 +5,41 @@ interface ShaResponse {
   sha: string;
 }
 
-export const API_URL_BASE = 'https://www.stoat.dev';
+export const STOAT_ORG = 'stoat-dev';
+export const INTERNAL_REPOS = ['stoat', 'stoat-action'];
+export const INTERNAL_REPO_DEFAULT_BRANCH = 'main';
+
+export const PROD_API_URL_BASE = 'https://www.stoat.dev';
+
+export const getApiUrlBase = async (ghOwner: string, ghRepo: string) => {
+  if (ghOwner !== STOAT_ORG || !INTERNAL_REPOS.includes(ghRepo)) {
+    return PROD_API_URL_BASE;
+  }
+
+  const branchName = core.getInput('pr_branch_name');
+  if (branchName === INTERNAL_REPO_DEFAULT_BRANCH) {
+    return PROD_API_URL_BASE;
+  }
+
+  const subdomain = branchName.replace(/[^-a-zA-Z0-9]/g, '-');
+  const devApiUrlBase = `https://stoat-git-${subdomain}-stoat-dev.vercel.app`;
+
+  try {
+    const response = await fetch(devApiUrlBase);
+    if (response.ok) {
+      return devApiUrlBase;
+    }
+    core.warning(`Testing connection to "${devApiUrlBase}" failed: ${response.status} - ${response.statusText}`);
+  } catch (e) {
+    core.warning(`Testing connection to "${devApiUrlBase}" failed: ${e}`);
+  }
+
+  core.warning(`Fall back from "${devApiUrlBase}" to ${PROD_API_URL_BASE}`);
+  return PROD_API_URL_BASE;
+};
 
 export async function waitForShaToMatch(repoSha: string) {
```

Comment:
Shouldn't this be used for dev servers and prod tests on internal repos to make sure it's testing against the correct commit? It looks like its usage was removed at some point and `PROD_API_URL_BASE` doesn't cover all cases.

Reply:
Yes, waiting for SHA works for actions triggered by the `stoat` project. However, it does not work for actions triggered by the `stoat-action` project, and the action project has no idea what the dev SHA should be on the server, which is only available in the `stoat` project.

I will add the SHA waiting mechanism back.